src/metabase/db/liquibase.clj

```diff
@@ -116,6 +116,35 @@
     (liquibase.h2/h2-database liquibase-conn)
     (.findCorrectDatabaseImplementation (DatabaseFactory/getInstance) liquibase-conn)))
 
+(when config/is-test?
+  ;; A lot of time during migration testing is spent parsing Liquibase migration YAMLs. The code below adds cache
+  ;; for the parsed files by implementing a custom caching parser.
+  (def ^:private ^java.util.concurrent.ConcurrentHashMap -test-only-migration-file-cache
+    (java.util.concurrent.ConcurrentHashMap.))
+  (def ^:private ^:dynamic -test-only-*return-caching-parser* true)
```

Comment:
> What do you think about putting these things in a metabase.test.liquibase namespace? That way we can keep all the code out of the actualy application.

Do you mean creating such a namespace? Because I couldn't find an existing one with this name. I'd love to stash it in some test-only part of the project, but I'm not yet aware of such place that would for sure be loaded early.

Reply:
Yeah, I mean creating a new namespace. I don't think there's anywhere better to enable the behavior, hence suggesting the `requiring-resolve` in this namespace still. This at least guarantees that the code can never be run in production, and avoids the need to use fancy prefixes.